### PR TITLE
add middlewares-user branch for user

### PR DESCRIPTION
**Pull request by HammamAbuShehadeh**
#15 add admin, user, and google authentication

### CODE REVIEW

**Comment by HammamAbuShehadeh** [comment]
Done Update scheme models files

File: server/server/middlewares/userAuth.js [hunk 1]
```diff
@@ -1 +1,54 @@
-// 
\ No newline at end of file
+/***********************************ramah */
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js]: remove unnecessary  comment

File: server/server/middlewares/userAuth.js [hunk 2]
```diff
@@ -1 +1,54 @@
...
+const userModel = require("../models/user");
+const blackListModel = require("../models/blackList");
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js]: should the name start with a capital letter

File: server/server/middlewares/userAuth.js [hunk 3]
```diff
@@ -1 +1,54 @@
...
+    };
+    
+    const cookies = cookie.parse(authHeader);
+    const accessToken = cookies["SessionID"];  
+
+    if (!accessToken) {
+     return res.sendStatus(401);
+   };
+   
+   const checkIfBlackListed = await blackListModel.findOne({ token:  accessToken});
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js]: should the name start with a capital letter for model name

File: server/server/middlewares/userAuth.js [hunk 4]
```diff
@@ -1 +1,54 @@
...
+     return res.status(401).json({ message: 'This session has expired. Please login'})
+   }
+
+   jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN, async (err, decoded) => {
+      if (err) {
+        return res.status(401).json({ message: "This session has expired. Please login" });
+      };
+
+      const { id } = decoded;
+      const user = await userModel.findById(id);
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js]: should the name start with a capital letter model name

File: server/server/middlewares/userAuth.js (hunk 1 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js]: Done

File: server/server/middlewares/userAuth.js (hunk 2 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js]: Done

File: server/server/middlewares/userAuth.js (hunk 3 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js]: Done

File: server/server/middlewares/userAuth.js (hunk 4 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js]: Done

File: server/server/middlewares/userAuth.js
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js:1]: looks good thx 
if there is another func you can add it

Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js:1]: looks good thx
if there is another func you can add it

File: server/server/middlewares/userAuth.js [hunk 5]
```diff
@@ -46,8 +46,19 @@ const ensureAuth = async (req, res, next) => {
...
+// For a user who's not logged in
+const ensureGuest = (req, res, next) => {
+  if (req.isAuthenticated()) {
+    res.redirect('/test');
+
+  } else {
+    return next();
+  };
+};
+
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js:58]: looks good

File: server/server/middlewares/userAuth.js [hunk 6]
```diff
@@ -56,9 +56,26 @@ const ensureGuest = (req, res, next) => {
...
+    if (role === 'user') {
+      return res.status(401).json({ message: "You are not authorized to view this page."});
+    };
+
+    next();
+
+  } catch (err) {
+    res.status(500).json({ message: err.message});
+  };
+};
```
Comment (HammamAbuShehadeh) [inline server/server/middlewares/userAuth.js:74]: looks good thx

**Review by HammamAbuShehadeh — COMMENTED** [review]
looks good thx
if there is another func you can add it

File: server/server/middlewares/userAuth.js
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js:1]: Ok

Comment (ramah-madi) [reply server/server/middlewares/userAuth.js:1]: Done

File: server/server/middlewares/userAuth.js (hunk 5 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js:58]: Thank you!

File: server/server/middlewares/userAuth.js (hunk 6 above)
Comment (ramah-madi) [reply server/server/middlewares/userAuth.js:74]: Ok

**Comment by ramah-madi** [comment]
All the functions have been added👍

**Pull request merged**